src/Lean/MetavarContext.lean

```diff
@@ -218,6 +218,10 @@ very simple unification and/or non-nested TC. So, if the "app builder" becomes a
 we may solve the issue by implementing `isDefEqCheap` that never invokes TC and uses tmp metavars.
 -/
 
+/--
+The LocalInstance stores information that the kernel does not care about, but is still
+important to carry for user-facing reasons about local instances in the LocalContext.
```

Comment:
This is not a good docstring. It does not explain what the structure is used for, only what it's not used for.

Reply:
I added an explanation, please do check if it's correct (I'm unsure my recollection of the explanation of how `LocalInstance` is used is accurate).